stumpy/stump_tiles.py

```diff
@@ -0,0 +1,784 @@
+
+    ρ : numpy.ndarray
+        The (top-k) Pearson correlations, sorted in ascending order per row
+
+    ρL : numpy.ndarray
+        The top-1 left Pearson correlations
+
+    ρR : numpy.ndarray
+        The top-1 right Pearson correlations
+
+    I : numpy.ndarray
+        The (top-k) matrix profile indices
+
+    IL : numpy.ndarray
+        The top-1 left matrix profile indices
+
+    IR : numpy.ndarray
+        The top-1 right matrix profile indices
+
+    ignore_trivial : bool
+        Set to `True` if this is a self-join. Otherwise, for AB-join, set this to
+        `False`. Default is `True`.
+
+    k : int
+        The number of top `k` smallest distances used to construct the matrix profile.
+        Note that this will increase the total computational time and memory usage
+        when k > 1.
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    `DOI: 10.1007/s10115-017-1138-x \
+    <https://www.cs.ucr.edu/~eamonn/ten_quadrillion.pdf>`__
+
+    See Section 4.5
+
+    The above reference outlines a general approach for traversing the distance
+    matrix in a diagonal fashion rather than in a row-wise fashion.
+
+    `DOI: 10.1145/3357223.3362721 \
+    <https://www.cs.ucr.edu/~eamonn/public/GPU_Matrix_profile_VLDB_30DraftOnly.pdf>`__
+
+    See Section 3.1 and Section 3.3
+
+    The above reference outlines the use of the Pearson correlation via Welford's
+    centered sum-of-products along each diagonal of the distance matrix in place of the
+    sliding window dot product found in the original STOMP method.
+    """
+    m_inverse = 1.0 / m
+    constant = (m - 1) * m_inverse * m_inverse  # (m - 1)/(m * m)
+    uint64_m = np.uint64(m)
+
+    # for storing diagonal iteration ranges
+    iter_ranges = np.zeros((config.STUMPY_N_DIAGONALS, 3), dtype=np.int64)
+    # for storing covariance values while iterating in chunks
+    covariances = np.zeros(config.STUMPY_N_DIAGONALS, dtype=np.float64)
+    # for storing range of diagonals to traverse in chunks
+    chunk_diags_range = np.arange(config.STUMPY_N_DIAGONALS)
+
+    for tile_idx in range(tiles_ranges[thread_idx, 0], tiles_ranges[thread_idx, 1]):
+        (
+            tile_i,  # i location of current tile in distance profile
+            tile_j,  # j location of current tile in distance profile
+            tile_height,  # height of current tile
+            tile_width,  # width of current tile
+            tile_start_diag,  # index of first diagonal to traverse on current tile
+            tile_stop_diag,  # index of last diagonal to traverse on current tile + 1
+            _,
+        ) = tiles[tile_idx]
+
+        # total number of diagonals to traverse
+        n_diags = tile_stop_diag - tile_start_diag
+        # number of diagonal "chunks" if we divide our diagonals into chunks
+        n_chunks = int(np.ceil(n_diags / config.STUMPY_N_DIAGONALS))
+
+        # chunk_idx is the index of the chunk we are dealing with
+        for chunk_idx in range(0, n_chunks):
```

Comment:
I somehow deleted my comment! I am going to write it again!

So, we can simplify the code by getting rid of `n_diags` and `n_chunks` that are computed just before the for-loop `for chunk_idx in range(...)`. 

So, instead of keeping track of `chunk_idx`, we can just jump over the diags with `step==config.STUMPY_N_DIAGONALS`, as follows:

```
chunk_diags_range = np.arange(config.STUMPY_N_DIAGONALS)
for tile_idx in range(...):
        (...) = tiles[tile_idx]
        for chunk_start in range(
            tile_start_diag, tile_stop_diag, config.STUMPY_N_DIAGONALS    # `step==config.STUMPY_N_DIAGONALS`
        ):
            chunk_size = min(config.STUMPY_N_DIAGONALS, tile_stop_diag - chunk_start)
            tile_diags = chunk_start + chunk_diags_range[:chunk_size]

            iter_ranges[:chunk_size, 0] = tile_diags - tile_i + tile_j
            ....
   
```

Reply:
Oh sick man this looks much better. Done!